app/lib/backend_api_logic/proxy_extension.rb

```diff
@@ -5,24 +5,30 @@ module ProxyExtension
     extend ActiveSupport::Concern
 
     included do
-      delegate :backend_api, to: :service, allow_nil: true
-      delegate :private_endpoint, :private_endpoint=, to: :backend_api, allow_nil: true, prefix: true
-      delegate :default_api_backend, to: 'BackendApi'
+      delegate :api_backend,       to: :service
+      delegate :api_backend=,      to: :service
+      delegate :backend_api,       to: :service
+      delegate :backend_api_proxy, to: :service
 
-      alias_method :api_backend, :backend_api_private_endpoint
-      alias_method :api_backend=, :backend_api_private_endpoint=
+      delegate :private_endpoint, :private_endpoint=, to: :backend_api, prefix: true
+      delegate :default_api_backend, to: 'BackendApi'
 
-      validates :backend_api, nested_association: {report: {private_endpoint: :api_backend}}, associated: true
       before_save :save_backend_api
 
       has_many :backend_api_configs, through: :service
       accepts_nested_attributes_for :backend_api_configs
+
+      validates :backend_api, nested_association: {report: {private_endpoint: :api_backend}}, associated: true, if: :validate_backend_api?
     end
 
     protected
 
+    def validate_backend_api?
+      backend_api.private_endpoint && backend_api.changed?
```

Comment:
Shhouldn't we check that what changed is the private_endpoint in particular?

Reply:
In fact, why should proxy know that it is private_endpoint that has changed?

I'd even say it should be something like:

```suggestion
      backend_api.validate_for_save?
```

```ruby
    def validate_for_save?
      if account.provider_can_use?(:api_as_product)
        # some condition
      else
        # some condition
     end    
    end
```

Will take a look tomorrow

Reply:
good point :+1: :muscle: